src/app/UploadPilets.tsx

```diff
@@ -1,23 +1,50 @@
-import { FC, ChangeEvent, useState } from 'react';
-import { CustomInput, InputGroup, InputGroupAddon, Button } from 'reactstrap';
+import { FC, ChangeEvent, useState, useEffect } from 'react';
+import { CustomInput, InputGroup, InputGroupAddon, Button, Progress, UncontrolledAlert } from 'reactstrap';
 import { jsx } from '@emotion/core';
 import { injectPiletsFromUrl } from './utils';
 
-export interface UploadPiletsProps {}
+export interface UploadPiletsProps { }
+
+const SucceededAlert = ({ piletName }) => {
+  const [open, setOpen] = useState(true);
+
+  useEffect(() => {
+    setOpen(true);
+    const timeoutId = setTimeout(() => setOpen(false), 5000);
+    return () => clearTimeout(timeoutId);
+  }, [piletName]);
+
+  return (
+    <UncontrolledAlert color="success" isOpen={open} style={{ marginTop: 10 }}>
+      The pilet '{piletName}' was added successfully!
+    </UncontrolledAlert>
+  );
+};
 
 export const UploadPilets: FC<UploadPiletsProps> = () => {
   const [file, setFile] = useState({
     value: undefined,
     key: 0,
   });
 
+  const [uploadData, setUploadData] = useState({
+    progress: 0,
+    piletName: undefined,
+    errMessage: undefined,
+  });
+
   const uploadPilet = (e: ChangeEvent<HTMLInputElement>) =>
     setFile({
       ...file,
       value: e.currentTarget.files[0],
     });
+
   const upload = () => {
     if (file) {
+      setUploadData({
+        ...uploadData,
```

Comment:
This does not work as you use a presumably outdated / captured version of `uploadData`. Use the dispatch overload instead: `setUploadData(data => ({ ...data, progress: 50 }))`.